prog/dftb+/lib_io/hsdutils2.F90

```diff
@@ -121,54 +122,80 @@ end subroutine getUnprocessedNodes
 
 
   !> Prints a warning message about unprocessed nodes
-  subroutine warnUnprocessedNodes(node, tIgnoreUnprocessed, nodeList)
+  subroutine warnUnprocessedNodes(node, ignoreList, nodeList)
 
     !> Root element of the tree to investigate
     type(fnode), pointer :: node
-
-    !> if anything left after processing should be flagged
-    logical, intent(in), optional :: tIgnoreUnprocessed
+    
+    !> list of left over nodes (if present)
+    type(fnodeList), pointer, optional :: ignoreList
 
     !> list of left over nodes (if present)
     type(fnodeList), pointer, optional :: nodeList
 
-    type(fnodeList), pointer :: list
-    type(fnode), pointer :: child
-    type(string) :: msg
-    integer :: ii, ll
-    logical :: tIgnoreUnprocessed0
+    type(fnodeList), pointer :: list, tmplist
+    type(fnode), pointer :: child, parent, childI
+    type(string) :: msg, name1,name2, nameI
+    integer :: ii, ll, jj, lI
+
+    call getUnprocessedNodes(node, tmplist)
+    ll = getLength(tmplist)
+    if (present(ignoreList)) then
+      lI = getLength(ignoreList)
+    else
+      lI = 0
+    end if    
+    nullify(list)
 
-    call getUnprocessedNodes(node, list)
-    ll = getLength(list)
     if (ll > 0) then
       msg = "The following " // i2c(ll) &
           &// " node(s) have been ignored by the parser:" // newline
       do ii = 0, ll-1
-        child => item(list, ii)
-        call append_to_string(msg, "(" // i2c(ii+1) // ")")
-        call appendPathAndLine(child, msg)
-        call append_to_string(msg, newline)
+        child => item(tmplist, ii)
+        parent => getParentNode(child)
+        call getNodeName(child, name1)
+        call getNodeName(parent, name2)
+        if (lI == 0) then
+          call append(list, child)    
+          call append_to_string(msg, "(" // i2c(ii+1) // ")")
+          call appendPathAndLine(child, msg)
+          call append_to_string(msg, newline)
+        end if  
+        do jj = 0, lI-1
+          childI => item(ignoreList, jj)
+          call getNodeName(childI, nameI)
+          ! check by name both node and parent (for block ignores)
+          if (char(name1) /= char(nameI) .and. char(name2) /= char(nameI)) then
+            call append(list, child)    
+            call append_to_string(msg, "(" // i2c(ii+1) // ")")
+            call appendPathAndLine(child, msg)
+            call append_to_string(msg, newline)
+          end if  
+        end do  
       end do
```

Comment:
We should just simply check, whether ``child`` has a given attribute set, and only include it in the list, it the attribute ``hsdIgnoreUnproc`` is not present or set to ``0``.